### DISCUSSION TITLE
Passing a JWT Supabase token to useShape

### DISCUSSION THREAD

**Discussion by barbudour** [electric-sql/electric#1538] [Questions]
Hi, I was able to start a sync service and read data from it using `useShape`. But now from the client I can read all records from the database, or with a certain selection through `where`. I have not found in the documentation a description of how I can pass a supabase token or a supabase client to JWT, so that I can get RLS configured in supabase to work.

**Comment by KyleAMathews** [comment]
Hey we're going to be adding an example of doing authentication and authorization soon — https://github.com/electric-sql/electric/issues/1455

Supporting RLS directly will be a bit tricky as Electric both queries data and gets updates from the replication log which doesn't use RLS. So we'd need to reproduce the logic there.

In the meantime what we suggest is you use an API to do authentication and authorization for your shape requests.

You can see an example of proxying here and we'll have a full example allowing/denying requests soon ish 

https://github.com/electric-sql/electric/blob/main/examples/nextjs-example/app/shape-proxy/%5B...table%5D/route.ts

**Reply by barbudour** [reply]
I have quite a bit of logic implemented in RLS. What options are possible to implement similar functionality to RLS?

**Reply by KyleAMathews** [reply]
You'd need to port the logic to your API. Could you give some specific examples of your RLS logic?

**Comment by Nycz-lab** [comment]
Hey @KyleAMathews,
any updates on this? Is there a way to get this to work in combination with RLS ?

**Reply by KyleAMathews** [reply]
No updates. You can still either port logic or query your DB to see if a user can access a shape
